htdocs/expedition/class/expedition.class.php

```diff
@@ -373,13 +410,14 @@ function create_line($entrepot_id, $origin_line_id, $qty)
 	 * Create the detail (eat-by date) of the expedition line
 	 *
 	 * @param 	object		$line_ext		full line informations
+	 * @param	array		$array_options		extrafields array
 	 * @return	int							<0 if KO, >0 if OK
 	 */
-	function create_line_batch($line_ext)
+	function create_line_batch($line_ext,$array_options=0)
 	{
 		$error = 0;
 
-		if ($this->create_line(($line_ext->entrepot_id?$line_ext->entrepot_id:'null'),$line_ext->origin_line_id,$line_ext->qty) < 0)
+		if ($this->create_line(($line_ext->entrepot_id?$line_ext->entrepot_id:'null'),$line_ext->origin_line_id,$line_ext->qty,$array_options) < 0)
 		{
 			$error++;
 		}
```

Comment:
Did you test a shipment of a productline with a lot managed product using a mysql database? 
It is in fact not an error in your PR, but a bad use of last_insert_id() in the lot module. One should not use it after a function (a function can insert in multple tables) but after an insert statement. I was also not aware of it before I noticed the error, when using a mysql db the first table name parameter is ignored and the last insert is used.
So I think we should adapt the create_line function to return the expeditiondet rowid if successful.